velodyne_pointcloud/src/conversions/pointcloudXYZIR.cc

```diff
@@ -19,13 +19,14 @@
 namespace velodyne_pointcloud
 {
 void PointcloudXYZIR::addPoint(
-  const float & x, const float & y, const float & z, const uint16_t & ring,
-  const uint16_t & azimuth, const float & distance, const float & intensity,
-  const double & time_stamp)
+  const float & x, const float & y, const float & z,
+  (void)const uint8_t & return_type, const uint16_t & ring, const uint16_t & azimuth,
```

Comment:
@taikitanaka3 
`(void)const uint8_t`　これはビルドできません。
`(void)` を消して下さい。

Reply:
```suggestion
  [[maybe_unused]]const uint8_t & return_type, const uint16_t & ring, const uint16_t & azimuth,
```

Reply:
@drwnz @miursh 
https://github.com/tier4/velodyne_vls/pull/26#discussion_r710703743
とありますが上記の[[]maybe_unused]]でよさそうでしょうか？

Reply:
@taikitanaka3 @miursh 
azimuth とかと同じなので、[[maybe_unused]]なしでいいと思います。
Overloaded functionのprototypeがpoint cloud typeによって変わらないようにあるので...

Reply:
@taikitanaka3 
[[maybe_unused]]をつけるのが今のスタイルとして良いんですかね？ 
付けても良いと思いますが、同じ並びの仮引数で付いているもの (return_type) と付いていないもの (azimuth, distance, time_stamp) とがあると誤解・混乱を生みそうだなと思って、付けないほうが良いというコメントをしました。
個人的には統一してあれば全部[[maybe_unused]]ありでも、全部なしでも大丈夫です。

Reply:
maybe_unused or void がないと不要な変数とみなされコンパイルが通りません。
そうですね、今のスタイルでは(void)よりも[[maybe_unused]]のほうが主流なきがしますが、
とりあえず差分の少ないvoidに統一しておきました。

Reply:
39a8adb

Reply:
@taikitanaka3 @miursh ありがとうございます！
@kenji-miyake の意見を聞きたいですが、
My understanding of [[maybe_unused]] is that it is only used in cases of conditional compilation or assertions when the variable actally may, or may not be used.
Because this is an overloaded function, it is not strictly correct to use [[maybe_unused]] but should probably employ un-named parameters.
I think the merged version is fine, but out of interest I'd like to hear Miyake-san's opinion here for best practice: un-named parameters in the function protoype, or as has been implemented here with `(void) const ...`? (英語になってしまってすみません）

Reply:
@kenji-miyake 
オーバーロードする関数は[[maybe_unused]]と(void)で対応するのどちらがベストプラクティスになるのでしょうか？ご意見いただけると助かります。

Reply:
@drwnz @taikitanaka3 
※この件はそこまで詳しくないので、正確ではないかもしれないことをご了承ください :bow: 

方法としては田中さんのだと1つ足りていなくて、以下の3つだと思います。
1. [[maybe_unused]]
2. unnamed parameter
3. (void) in function definition

まず3は「使っていない変数を無理やり使っていることにする」ためお行儀が良くないので、1, 2が使えない場合に限るのが良いかと。
1, 2はどっちでも良い気はしますが、「DEBUGビルド時にだけ使う変数がある」ような場合に2は使えないので、1に統一しておくのが無難ではないでしょうか？ :thinking: 
確実に変数を使わないと分かっている場合は、2にしても良いと思います。

参考
https://zenn.dev/tetsurom/books/cpp11-cpp20-for-game-developers/viewer/007-maybe-unused
https://stackoverflow.com/questions/61740734/c-warning-virtual-function-override-with-unused-arguments

Reply:
@kenji-miyake 
今後の参考になります！
@miursh 
> https://github.com/tier4/velodyne_vls/pull/26#discussion_r718114925
こちらに関しては[[maybe_unused]]が良さそうですね。

Reply:
@kenji-miyake ありがとうございます！ (勉強のためだけのディスカッションですので、違うところで話したほうがいいなら教えて下さい）
Conditiional compilationでunusedになるケースではないので、[[maybe_unused]]じゃなくて、definitely unusedですね。その場合はunamedの方がいい気がします。Code readabilityを考えると、[[maybe_unused]]だとどういうケースだとusedになるを探しますが、unamedだとこのoveridden functionの場合は、使われてない変数とすぐ分かります。
このケースで、関数は使われてないのでDEBUGビルドでも使わないです。
C++17から[[maybe_unused]]がある理由は、conditional compilationのケースなので、C++11でも対応するun-named parameterが使えるところで使うとあまり良くない気がします。
どうでしょうか？

Reply:
@drwnz stackoverflowでもそのように書いてある気がしますが、正直人によると思います。
現状unusedなのでunusedが正しいというのは事実です。

ただし、今後デバッグしたい人が現れた時、`[[maybe_unused]]`だとその人が単にデバッグコードを書けば良いのに対して、`unnamed parameter`だとパラメータ名の記載からしないといけません。
その時、若干名前を調べる手間がかかったり、調べずに適当に書いた結果、他のオーバーライド関数とは違う名前で定義してしまって奇妙なコードになる、のようなリスクも有ります。

また、`[[maybe_unused]]`の場合、（正しい使い方か分かりませんが）「オーバーライド関数次第で使ったり使わなかったりする」という意味を込めて、全てのオーバーライド関数に、`[[maybe_unused]]`を付けることもできます。
（その場合、本来使うべき値を使っていなかったのに見逃してしまう、みたいなリスクも有り得ますが…）
これは、オーバーライド毎に使う/使わないを判断して個別に`unnamed parameter`に変えるより楽だと思います。

要は、現時点での正しさに以外にも、将来的な変化等の側面も考慮する必要があるということです。
その様々な要素を共同開発者で議論して合意できたものを選択できていれば問題ないと思います。

（というはっきりしない回答なんですが、これで良いでしょうか？ :thinking: まぁ、細かい違いだと思うので、あまり時間を使うところでは無いかなという感じはします。）

Reply:
@kenji-miyake ありがとうございます！マイナーなポイントだと思いますので時間使うところではないけど、使ったことないのでC++17の勉強にもなり、ソフトウェア開発プロの三宅さんの意見をもらうことは勉強になるチャンスなのでありがたいです

Reply:
（僕もにわかなので、マニアックな話は山崎さんとかも含めて聞いてみると良いですね！これからもたくさん議論していきましょう！）